### PR TITLE
feat:(726) card composable

### PR DESCRIPTION
**Pull request by mutebg**
#726

**What**

1. Background - implement CardComposable 
2. What did you do - Added a few new components: CardComposable, CardMedia, CardContent, CardLink and CardActions
3. What does the reviewers should expect

**I have done:**
- [x] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [x] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by pp-serviceaccount** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs-pr.s3-website-eu-west-1.amazonaws.com/726-card-composable
  Storybook: http://ncu-newskit-docs-pr.s3-website-eu-west-1.amazonaws.com/726-card-composable/storybook

File: src/card-composable/card-composable.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+const ThemelessCardComposable = React.forwardRef<
+  HTMLDivElement,
+  CardComposableProps
+>(({children, areas = defaultAreas, ...props}, ref) => {
+  const overrides = useGetOverrides<CardComposableProps>(
+    props,
+    'cardComposable',
+  );
+
+  console.log({overrides});
```
Comment (mstuartf) [inline src/card-composable/card-composable.tsx:54]: console.log

File: src/card-composable/card-composable.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    const {useAreas} = useCardContext();
+    const overrides = useGetOverrides<CardComposableProps>(props, 'cardMedia');
+    return (
+      <StyledMedia
+        ref={ref}
+        areaName={useAreas ? 'media' : undefined}
+        overrides={overrides}
+        {...props}
+      >
+        {children || <Image alt="" {...media} />}
```
Comment (mstuartf) [inline src/card-composable/card-composable.tsx:81]: We could add a default `alt` - e.g. `card image`

File: src/card-composable/styled.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import {GridLayout} from '../grid-layout/grid-layout';
+import {CardLinkProps, StylableGridLayout} from './types';
+import {LinkStandalone} from '../link';
+
+type StyledGridLayoutProps = StylableGridLayout & {
+  areaName?: string;
+};
+
+const StyledGrid = styled(GridLayout)<StyledGridLayoutProps>`
+  ${getStylePreset('', '')};
```
Comment (mstuartf) [inline src/card-composable/styled.tsx:12]: How does this work when the first argument is `''`?

File: src/grid-layout/__tests__/stories/grid-card.tsx [hunk 4]
```diff
@@ -175,3 +175,18 @@ export const GridTeaser = ({
     </StyledAdvancedCard>
   );
 };
+
+// const hasTouchLink = (children: any, href: string): boolean =>
```
Comment (mstuartf) [inline src/grid-layout/__tests__/stories/grid-card.tsx:179]: Should this be removed?

File: src/card-composable/__tests__/__stories/card-composable.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,947 @@
...
+        <H />
+        <P />
+      </CardContent>
+
+      <CardMedia
+        media={{src: 'https://storybook.newskit.co.uk/placeholder-3x2.png'}}
+      />
+
+      <CardActions>
+        <Tag href="/news">Tag</Tag>
```
Comment (mstuartf) [inline src/card-composable/__tests__/__stories/card-composable.stories.tsx:167]: I don't know if we have an agreed pattern for this but I've been disabling hrefs in stories using `href="javascript:void(0);"` so the user doesn't get an ugly 404 on click. Don't feel strongly though.

**Comment by mstuartf** [comment]
Looks good!

File: src/card-composable/card-composable.tsx (hunk 2 above)
Comment (mutebg) [reply src/card-composable/card-composable.tsx:81]: no really, that would be wrong from an accessibility point of view. Better to be empty or for consumers to provide something that's useful

File: src/card-composable/styled.tsx (hunk 3 above)
Comment (mutebg) [reply src/card-composable/styled.tsx:12]: it ignores it. The data is passes always as 2nd the argument. that happens in `useGetOverrides` which takes the default values and merges them with the overrides prop. We do this "pattern" in few more places.

File: src/card-composable/__tests__/__stories/card-composable.stories.tsx (hunk 5 above)
Comment (mutebg) [reply src/card-composable/__tests__/__stories/card-composable.stories.tsx:167]: good one, will update all stories

File: src/grid-layout/__tests__/stories/grid-card.tsx (hunk 4 above)
Comment (mutebg) [reply src/grid-layout/__tests__/stories/grid-card.tsx:179]: I think so, removed it. tnx

File: src/card-composable/card-composable.tsx (hunk 2 above)
Comment (mstuartf) [reply src/card-composable/card-composable.tsx:79]: ok

**Review by JohnTParsons — APPROVED** [review]

**Pull request merged**